scripts/reboot_helper.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python3
+#
+# reboot_helper.py
+#
+# Utility helper for reboot within SONiC
+
+import sonic_platform
+from sonic_py_common import logger
+from utilities_common.chassis import is_smartswitch, get_dpu_list
+
+SYSLOG_IDENTIFIER = "reboot_helper"
+
+EXIT_FAIL = -1
+EXIT_SUCCESS = 0
+
+# Global logger instance
+log = logger.Logger(SYSLOG_IDENTIFIER)
+
+# Global variable for platform chassis
+platform_chassis = None
+
+
+def load_platform_chassis():
+    """
+    Load the platform chassis using the SONiC platform API.
+
+    This function attempts to instantiate the platform chassis object.
+    If successful, it sets the global variable `platform_chassis` to the instantiated object.
+
+    Returns:
+        bool: True if the platform chassis is successfully loaded, False otherwise.
+    """
+    global platform_chassis
+
+    # Load new platform API class
+    try:
+        platform_chassis = sonic_platform.platform.Platform().get_chassis()
+        if platform_chassis is None:
+            log.log_error("Platform chassis is not loaded")
+            return False
+        return True
+    except Exception as e:
+        log.log_error("Failed to instantiate Chassis: {}".format(str(e)))
+        return False
+
+
+def load_and_verify(module_name):
+    """
+    Load the platform chassis and verify the required parameters.
+
+    Args:
+        module_name (str): The name of the module to verify.
+
+    Returns:
+        bool: True if platform chassis is successfully loaded and required parameters are verified, False otherwise.
+    """
+    # Load the platform chassis if not already loaded
+    if not load_platform_chassis():
+        return False
+
+    if not is_smartswitch():
+        log.log_error("Platform is not a smartswitch")
+        return False
+
+    dpu_list = get_dpu_list()
+    if module_name.lower() not in dpu_list:
+        log.log_error("Module {} not found".format(module_name))
+        return False
+
+    return True
+
+
+def reboot_dpu(module_name, reboot_type):
+    """
+    Reboot the specified module by invoking the platform API.
+
+    Args:
+        module_name (str): The name of the module to reboot.
+        reboot_type (str): The type of reboot requested for the module.
+
+    Returns:
+        bool: True if the reboot command was successfully sent, False otherwise.
+    """
+    if not load_and_verify(module_name):
+        return False
+
+    # Attempt to reboot the module
+    if not hasattr(platform_chassis, 'reboot'):
```

Comment:
the module reboot is not part of the chassis attribute, it is for each module, we need to obtain modules using `get_all_modules` and then perform the reboot

Reply:
I have obtained module_index and calling accordingly.